### ISSUE TITLE
Use swagger (Swashbuckle ) with ASP.NET core 2.2 and IIS virtual applications

### ISSUE THREAD

**Issue by EnricoMassone** [dotnet/aspnetcore#10514]
Hi, 

I'm currently testing the integration between ASP.NET core and swagger by using the `Swashbuckle.AspNetCore` nuget package.

My current test environment is the following:

- ASP.NET core 2.2 web application with one api controller
- the development machine is a windows 10 machine
- the installed version of IIS is IIS 10

I'm testing my application by using a virtual application under the IIS default web site. The virtual application has an alias of `test-swagger`. By following the documentation I configured the swagger middleware in the following manner (**notice the . (dot) character at the beginning of the swagger endpoint**): 

```
app.UseSwaggerUI(c =>
      {
        c.SwaggerEndpoint("./swagger/v1/swagger.json", "My API V1");
      });
```

By configuring swagger this way the swagger UI does **not** work as expected, a pop-up is showed when the path `/swagger` is called and the message showed states that the swagger UI is not able to find the swagger.json file. 

By following [this blog](https://bartwullems.blogspot.com/2018/08/aspnet-coreswagger-error-when-using.html) I finally solved my issue. Basically, the only difference is that **two . (dot) characters are required in the swagger endpoint configuration** (while the official docs reports that just one dot character must be used in the swagger endpoint as a prefix). 

To summarize, the working configuration for me is the following (**notice the two . (dot) characters at the beginning of the swagger endpoint**): 

```
app.UseSwaggerUI(c =>
      {
        c.SwaggerEndpoint("../swagger/v1/swagger.json", "My API V1");
      });
```

Is there an error in the official docs ? Am I missing something ? 

Thanks for helping !

**Comment by anurse** [comment]
It sounds like Swashbuckle isn't properly using the `HttpRequest.PathBase` value. Can you provide a sample project that reproduces the problem? What is the full absolute URL you expect for the swagger endpoint?

**Comment by EnricoMassoneDeltatre** [comment]
> It sounds like Swashbuckle isn't properly using the `HttpRequest.PathBase` value. Can you provide a sample project that reproduces the problem? What is the full absolute URL you expect for the swagger endpoint?

Hi and apologize for the late reply, 

the full absolute URL to which I would like to respond is something like `http://localhost/test-swagger/swagger/index.html` where `test-swagger` is the alias assigned to the application registered under the existing web site in IIS

**Comment by anurse** [comment]
It sounds like the issue is in Swashbuckle then, or possibly the doc. I'm surprised that you need `./` at all and isn't handling the path base properly itself.

**Comment by EnricoMassoneDeltatre** [comment]
> It sounds like the issue is in Swashbuckle then, or possibly the doc. I'm surprised that you need `./` at all and isn't handling the path base properly itself.

I totally agree with you. In my scenario, the only way to get it working properly is by using the prefix `../`

**Comment by anurse** [comment]
Have you filed an issue on the [Swashbuckle repository](https://github.com/domaindrivendev/Swashbuckle.AspNetCore)?

**Comment by EnricoMassoneDeltatre** [comment]
> Have you filed an issue on the [Swashbuckle repository](https://github.com/domaindrivendev/Swashbuckle.AspNetCore)?

I'll try to open an issue directly to Swashbuckle, by referencing this issue. I'll write here in case of news

**Comment by bprodduturi** [comment]
Even after trying with the double dot prefix, when I pull up http://localhost/myAspNetCore22App/swagger/index.html, I get a blank page. The browser console reports 404 on swagger-ui.css, swagger-ui-bundle.js, swagger-ui-standalone-preset.js.

**Comment by einaradolfsen** [comment]
Should the docs mention to use ../ instead, to save developers from troubleshooting?
